src/misc.jl

```diff
@@ -395,17 +395,17 @@ end
 discretize_make_ia(values::AbstractVector, levels) =
     IndirectArray(Array{UInt8}(indexin(values, levels)), levels)
 discretize_make_ia(values::AbstractVector)         = discretize_make_ia(values, unique(values))
-discretize_make_ia(values::AbstractVector, ::Void) = discretize_make_ia(values)
+discretize_make_ia(values::AbstractVector, ::Nothing) = discretize_make_ia(values)
 
 discretize_make_ia(values::IndirectArray)         = values
-discretize_make_ia(values::IndirectArray, ::Void) = values
+discretize_make_ia(values::IndirectArray, ::Nothing) = values
 
 discretize_make_ia(values::CategoricalArray) =
     discretize_make_ia(values, intersect(push!(levels(values), missing), unique(values)))
-discretize_make_ia(values::CategoricalArray, ::Void) = discretize_make_ia(values)
+discretize_make_ia(values::CategoricalArray, ::Nothing) = discretize_make_ia(values)
 function discretize_make_ia(values::CategoricalArray{T}, levels::Vector) where {T}
-    mapping = coalesce.(indexin(CategoricalArrays.index(values.pool), levels), 0)
-    unshift!(mapping, coalesce(findfirst(ismissing, levels), 0))
+    mapping = Union{Nothing,Int}[coalesce.(indexin(CategoricalArrays.index(values.pool), levels), 0)...]
+    pushfirst!(mapping, coalesce(findfirst(ismissing, levels), 0))
```

Comment:
The `coalesce` here should be `something`, since `findfirst` will return `nothing` rather than `missing`. The same likely applies above. (Also `indexin` might be deprecated, can't quite recall.)

Reply:
Also AFAICT the comprehension isn't necessary.